src/components/bar.cpp

```diff
@@ -656,8 +657,25 @@ void bar::handle(const evt::motion_notify& evt) {
   std::lock_guard<std::mutex> guard(m_mutex, std::adopt_lock);
 
   m_log.trace("bar: Detected motion: %i at pos(%i, %i)", evt->detail, evt->event_x, evt->event_y);
-#if WITH_XCURSOR
   m_motion_pos = evt->event_x;
+
+  tags::action_t m_hover_act_idx = m_action_ctxt->has_action(mousebtn::HOVER, m_motion_pos);
+  string m_hover_act = "";
+  if (m_hover_act_idx != tags::NO_ACTION) {
+    m_hover_act = m_action_ctxt->get_action(m_hover_act_idx);
+    m_log.trace("bar: Detected hover action %s", m_hover_act);
+  }
+  if (!string_util::compare(m_hover_act, m_last_hover_act)) {
+    m_log.trace("bar: Switched hover action from '%s' to '%s'", m_last_hover_act, m_hover_act);
+    if(m_last_hover_act != "") {
+      m_sig.emit(button_press{m_last_hover_act});
```

Comment:
Because hover works as a toggle. A hover action lets the module know that its hover state changed, it's up to the module to keep track of that state. I think that's the simplest way to handle it.

Reply:
Speaking of, I'm having a big problem: you can't emit button press signals twice in the same method, the controller swallows the second one. I'm sure there's a simple fix, but I'm not very familiar with the codebase yet.
@patrick96 could you give me a hand? Is there a simple way to enqueue the second signal? Or a worse alternative, emit one and then stall until the next iteration of the eventqueue worker loop?
For reference, this isn't a problem when going from hovering a module to hovering nothing, because there is no second action to be called, but it is when having two modules with hover next to each other.

Reply:
> Because hover works as a toggle. A hover action lets the module know that its hover state changed, it's up to the module to keep track of that state. I think that's the simplest way to handle it.

I see. In that case, do you think it would make more sense to have separate "mouse enter" and "mouse leave" events?

> Speaking of, I'm having a big problem: you can't emit button press signals twice in the same method, the controller swallows the second one.

Yeah, the enqueueing of commands isn't great. Currently we store the command in `m_inputdata` and swallow all other commands as long as it has not been handled and the variable reset.
This could probably easily be replaced with a queue of strings where we place incoming commands at the end and `process_inputdata` takes one command (or all of them) from the front and processes them.

Reply:
Yeah that's a good idea, but I think it's a bit outside of my scope.
I'll open an issue requesting that rework and put this on hold.

> I see. In that case, do you think it would make more sense to have separate "mouse enter" and "mouse leave" events?

I'm not sure. The problem with that would be that then the action would change. It can be taken into consideration, but I don't think the extra complexity it would need is worth it.

Reply:
> I'm not sure. The problem with that would be that then the action would change.

I don't think this would be very complex to solve. Right now we emit button presses when the current hover action changes. One for the new one and one for the old one. We would do the same thing but store both the current "mouse enter" action (so that we don't run on every mouse move, only the first time) and the current "mouse leave" action (so that we can detect when we leave a region and have to run the old leave action).

I'm just thinking about the usability. It's one thing if it's just the module itself that's keeping state, that's really easy. But as soon as the user wants to run something only while in the hover region, keeping state becomes very cumbersome.